src/org/rascalmpl/library/util/PathConfig.java

```diff
@@ -535,15 +545,17 @@ public static PathConfig fromSourceProjectRascalManifest(ISourceLocation manifes
                 }
 
                 if (libProjectName != null) {
-                    if (reg.exists(projectLoc) && dep != rascalProject) {
+                    if (reg.exists(projectLoc)) {
                         // The project we depend on is available in the current workspace. 
                         // so we configure for using the current state of that project.
                         PathConfig childConfig = fromSourceProjectRascalManifest(projectLoc, mode);
 
                         switch (mode) {
                             case INTERPRETER:
                                 srcsWriter.appendAll(childConfig.getSrcs());
-                                libsWriter.append(childConfig.getBin());
+                                if (dep != rascalProject) { // Never load Java classes from a non-current `rascal`
```

Comment:
what fails if you do load java classes from a rascal project? as in, the class loader will most likely ignore them? but it has less special cases to this code?

Reply:
What happens is that there will be two sources for all runtime classes including vallang and the interpreter and the library components written in Java. Creating new classes might work while adding or changing methods will not. Most likely if the two versions start to become different there will be ClassNotFound and such exceptions thrown by the JavaBridge. 

What I find most disturbing is that it becomes much less transparant what the linkage is between Rascal code and Java code to the programmer. 

One of the consequences of loading from the latest target folder of the rascal project is failing to bootstrap the tutor due to the above exceptions. Also many tests will fail.